core/build/src/lib.rs

```diff
@@ -69,13 +69,30 @@ pub fn sgx_builder() -> Builder {
         .ctypes_prefix("core::ffi")
         .allowlist_recursively(false)
         .parse_callbacks(Box::new(SgxParseCallbacks::default()))
+        .clang_args(env_c_flags())
         .clang_arg(&format!("-I{}", include_path));
 
     cargo_emit::rerun_if_changed!(include_path);
 
     builder
 }
 
+// Gets the `CFLAGS` from the environment, if any.  When there are no `CLFAGS`
+// will return an empty vector.
+// The `CFLAGS` will be split on whitespace in order to allow for multiple
+// arguments. This does *not* attempt to handle escaped shell characters,
+// https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html
```

Comment:
Up for suggestions on a better way to handle getting system headers to custom target triples, in particular `time.h` for `time_t`. 

I did find https://docs.rs/shlex/latest/shlex/ but didn't want to invest too much in this `CFLAGS` path if we think there may be a better way.

Reply:
I think `shlex` may be a useful addition to `mc-util-build-script::Environment` once we're in a place to move that to it's own crate.

Reply:
I'm interpreting this that we're fine for whitespace here at this time, but we shouldn't shy away from shlex going forward.

leaving as is unless I hear otherwise.